docs-src/security-model.md

```diff
@@ -0,0 +1,148 @@
+
+* Which namespaces can contain Gateways of the specified GatewayClass.
+* Which namespaces Routes can be targeted in by Gateways of the specified
+  GatewayClass.
+
+## Roles
+For the purposes of this security model, 3 common roles have been identified:
+
+* **Infrastructure provider**: The infrastructure provider (infra) is
+  responsible for the overall environment that the cluster(s) are operating in.
+  Examples include: the cloud provider (AWS, Azure, GCP, ...), the PaaS provider
+  in a company.
+* **Cluster operator**: The cluster operator (ops) is responsible for
+  administration of entire clusters. They manage policies, network access,
+  application permissions.
+* **Application developer**: The application developer (dev) is responsible for
+  defining their application configuration (e.g. timeouts, request
+  matching/filter) and Service composition (e.g. path routing to backends). 
+
+Although these roles can cover a wide variety of use cases, some organizations
+may be structured slightly differently. Many organizations may also have a
+fourth role that sits between "cluster operator" and "application developer":
+
+* **Application admin**: The application admin has administrative access to some
+  namespaces within a cluster, but not the cluster as a whole.
+
+## The Security Model
+There are two primary components to the Service APIs security model: RBAC and
+namespace restrictions.
+
+## RBAC
+RBAC (role-based access control) is the standard used for Kubernetes
+authorization. This allows users to configure who can perform actions on
+resources in specific scopes. RBAC can be used to enable each of the roles
+defined above. In most cases, it will be desirable to have all resources be
+readable by most roles, so instead we'll focus on write access for this model.
+
+### Write Permissions for Simple 3 Tier Model 
+| | GatewayClass | Gateway | Route | TrafficSplit |
+|-|-|-|-|-|
+| Infrastructure Provider | Yes | Yes | Yes | Yes |
+| Cluster Operators | No | Yes | Yes | Yes |
+| Application Developers | No | No | Yes | Yes |
+
+### Write Permissions for Advanced 4 Tier Model 
+| | GatewayClass | Gateway | Route | TrafficSplit |
+|-|-|-|-|-|
+| Infrastructure Provider | Yes | Yes | Yes | Yes |
+| Cluster Operators | Sometimes | Yes | Yes | Yes |
+| Application Admins | No | In Specified Namespaces | In Specified Namespaces | In Specified Namespaces |
+| Application Developers | No | No | In Specified Namespaces | In Specified Namespaces |
+
+## Namespace Restrictions
+The extra configuration options are not possible to control with RBAC. Instead,
+they will be controlled with configuration fields on GatewayClasses:
+
+* **gatewayNamespaceSelector**: This field is a selector of valid namespaces
+  that Gateways can use this GatewayClass from. An empty selector (default)
+  indicates that Gateways can use this GatewayClass from any namespace.
+
+* **multiNamespaceRoutes**: This field is a selector of valid namespaces that
```

Comment:
Did we talk about having this as a Gateway property? It seems like it would also make sense for the Gateway owner (whether they are the Cluster operator, or Application admin)  to be able to set what namespaces the routes come from, rather than having the GatewayClass owner needing to configure that.

Reply:
Yeah this is a tough one. If we don't have some kind of attribute/control at this level, anyone that can create a Gateway can reference Routes in any namespace. I'd imagine that at least some organizations with multi-tenant clusters would like to allow developers to create app-specific Gateways that can't access routes in other namespaces. 

I'm not sure if this is the source of your comment, but the name of this attribute when you commented was wrong, I've switched it back to `routeNamespaceSelector` which may sound more familiar. This `multiNamespaceRoutes` name was actually from an alternative that I'd considered. I've covered that potential approach in the "Alternatives Considered" section.

I'm open to any approach here, but I do think it would be valuable to have some way for admins to specify that certain Gateways can not reference all Routes in a cluster.

Reply:
Okay, so you're saying that in this case, this is the largest set of possible namespaces that Routes _could_ be referenced from. Whether we add a thing to Gateway to perform a similar function is a separate discussion? If so, that sounds okay to me.

Reply:
I've updated the docs around this now, hopefully this is a bit clearer now.